gym/utils/passive_env_checker.py

```diff
@@ -16,47 +18,33 @@ def _check_box_observation_space(observation_space: spaces.Box):
     if len(observation_space.shape) == 3:
         if observation_space.dtype != np.uint8:
             logger.warn(
-                f"It seems that your observation space ({observation_space}) is an image but the `dtype` of your observation_space is not `np.uint8`. "
-                "If your observation is not an image, we recommend you to flatten the observation to have only a 1D vector"
+                f"It seems a Box observation space is an image but the `dtype` is not `np.uint8`, actual type: {observation_space.dtype}. "
+                "If the Box observation space is not an image, we recommend flattening the observation to have only a 1D vector."
             )
-        if np.any(observation_space.low != 0) or np.any(
-            observation_space.high != 255
-        ):  # todo np.all?
+        if np.any(observation_space.low != 0) or np.any(observation_space.high != 255):
             logger.warn(
-                "It seems that your observation space is an image but the upper and lower bounds are not in [0, 255]. "
+                "It seems a Box observation space is an image but the upper and lower bounds are not in [0, 255]. "
                 "Generally, CNN policies assume observations are within that range, so you may encounter an issue if the observation values are not."
             )
 
     if len(observation_space.shape) not in [1, 3]:
```

Comment:
Maybe, however we mostly assume that all observation need to be compatible with neural networks that without additional pre-processing would require a vector (1) or image for convolutional with 3. 
This will only be raise with the user calls `check_env`